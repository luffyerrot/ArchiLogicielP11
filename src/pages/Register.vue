<template>
    <ConnexionForm
        :register="true"
    />
</template>
  
<script setup lang="ts">
</script>