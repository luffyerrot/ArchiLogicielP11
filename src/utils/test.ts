import { ComponentMountingOptions, mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { expect, it } from "vitest";
import { createVuetify } from "vuetify";
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export function factory(originalComponent: any, shallow = false, pinia = false, options?: ComponentMountingOptions<any, unknown>): VueWrapper<unknown, any> {
    if (pinia) {
        setActivePinia(createPinia());
    }
    if (shallow) {
        return shallowMount(originalComponent, options);
    } else {
        const vuetify = createVuetify({
            components,
            directives,
        });
        const defaultOptions = {
            global: {
                plugins: [vuetify]
            },
            ...options
        };
        return mount(originalComponent, defaultOptions);
    }
};

export function renderCorrectly(wrapper: VueWrapper<unknown, any>) {
    it('Bon affichage de la page', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
}