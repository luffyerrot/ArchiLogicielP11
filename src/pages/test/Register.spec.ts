import { factory, renderCorrectly } from "@/utils/test";
import { describe } from "vitest";
import Register from "../Register.vue";

describe('Page d\'enregistrement', () => {
    const wrapper = factory(Register, true);
    renderCorrectly(wrapper);
})