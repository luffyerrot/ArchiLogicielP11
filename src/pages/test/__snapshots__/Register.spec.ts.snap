// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page d'enregistrement > Bon affichage de la page 1`] = `"<connexion-form-stub register="true"></connexion-form-stub>"`;
